### PR TITLE
Document gmail smtp configuration

### LINKED ISSUE #3039: Extend mail client to support Gmail SMTP server
This issue covers:

* the test of the mail client with Gmail SMTP server
* the update of the guide to explain the required configuration

Requires Vert.x 3.8

### PR DESCRIPTION
**Pull request by Dufgui**
fix #3039

### CODE REVIEW

File: docs/src/main/asciidoc/sending-emails.adoc [hunk 1]
```diff
@@ -246,6 +246,36 @@ class MailTest {
 }
 ----
 
+== Gmail specific configuration
+
+If you want to use the gmail smtp server, first, please create a dedicated password in `google account > security > application password` or go to https://myaccount.google.com/apppasswords
```
Comment (gsmet) [inline docs/src/main/asciidoc/sending-emails.adoc]:
```suggestion
If you want to use the Gmail SMTP server, first create a dedicated password in `Google Account > Security > App passwords` or go to https://myaccount.google.com/apppasswords.
```

File: docs/src/main/asciidoc/sending-emails.adoc [hunk 2]
```diff
@@ -246,6 +246,36 @@ class MailTest {
 }
 ----
 
+== Gmail specific configuration
+
+If you want to use the gmail smtp server, first, please create a dedicated password in `google account > security > application password` or go to https://myaccount.google.com/apppasswords 
+
+When it done you can use it in your configuration.
```
Comment (gsmet) [inline docs/src/main/asciidoc/sending-emails.adoc]:
```suggestion
When done, you can configure your Quarkus application by adding the following properties to your `application.properties`: 
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks for taking the time to test that.

I added some wording suggestions, could you include them and squash your commits in one?

Thanks!

File: docs/src/main/asciidoc/sending-emails.adoc [hunk 3]
```diff
@@ -246,6 +246,36 @@ class MailTest {
...
+----
+
+Or with SSL:
+
+[source]
+----
+quarkus.mailer.from=YOUREMAIL@gmail.com
+quarkus.mailer.host=smtp.gmail.com
+quarkus.mailer.port=465
+quarkus.mailer.ssl=true
```
Comment (machi1990) [inline docs/src/main/asciidoc/sending-emails.adoc]: this will work only in combination with `quarkus.ssl.native=true` .

Comment (Dufgui) [reply docs/src/main/asciidoc/sending-emails.adoc]: I can add it in the doc but I propose to change `SslNativeConfigBuildItem` as a `MultiBuildItem` and produce them in the `MailerProcessor` if the `MailConfig.ssl` is enable ? Are you ok @gsmet @machi1990  ?
Else I just document it....

Comment (gsmet) [reply docs/src/main/asciidoc/sending-emails.adoc]: I added a note about the SSL support and will merge.

I wonder if we should enable SSL by default when the Mailer extension is used. @cescoffier WDYT?

**Review by gsmet — APPROVED** [review]

**Pull request merged**